File: MyPortal.Core/Entities/Document.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MyPortal.Core.Interfaces;
using QueryKit.Repositories.Attributes;

namespace MyPortal.Core.Entities
{
    [Table("Documents")]
    public class Document : Entity, IAuditableEntity, IDirectoryEntity, ISoftDeleteEntity, IVersionedEntity
    {
        public Guid TypeId { get; set; }

        public Guid DirectoryId { get; set; }

        [Required, StringLength(512)] 
        public string StorageKey { get; set; } = null!;

        [Required, StringLength(256)] 
        public string FileName { get; set; } = null!;

        [Required, StringLength(256)] 
        public string ContentType { get; set; } = null!;

        public long? SizeBytes { get; set; }

        [StringLength(128)]
        public string? Hash { get; set; }

        [Required, StringLength(128)]
        public string? Title { get; set; }
        
        [StringLength(256)]
        public string? Description { get; set; }

        // Only visible to staff users who have access to the directory
        public bool IsPrivate { get; set; }

        [SoftDelete]
        public bool IsDeleted { get; set; }

        public Directory? Directory { get; set; }

        public DocumentType? Type { get; set; }
        
        public Guid CreatedById { get; set; }
        public string CreatedByIpAddress { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public Guid LastModifiedById { get; set; }
        public string LastModifiedByIpAddress { get; set; } = string.Empty;
        public DateTime LastModifiedAt { get; set; }
        public User? CreatedBy { get; set; }
        public User? LastModifiedBy { get; set; }
        public long Version { get; set; }
    }
}

File: MyPortal.Frontend/myportal-spa/src/app/shared/types/document.ts
export interface DocumentDetailsResponse {
  id: string;
  typeId: string;
  typeDescription: string;
  directoryId: string;
  createdById: string;
  createdByName: string;
  createdAt: string;
  lastModifiedById: string;
  lastModifiedByName: string;
  lastModifiedAt: string;
  storageKey: string;
  fileName: string;
  contentType: string;
  sizeBytes?: number | null;
  hash?: string | null;
  title?: string | null;
  description?: string | null;
  isPrivate: boolean;
  isDeleted: boolean;
}

export interface DirectoryDetailsResponse {
  id: string;
  name: string;
  parentId?: string | null;
  isPrivate?: boolean;
  uploadPolicy?: number;
}

export interface DirectoryContentsResponse {
  directory: DirectoryDetailsResponse;
  directories: DirectoryDetailsResponse[];
  documents: DocumentDetailsResponse[];
}

export interface DirectoryTreeResponse {
  directory: DirectoryDetailsResponse;
  directories: DirectoryTreeResponse[];
  documents: DocumentDetailsResponse[];
}

export const OTHER_DOCUMENT_TYPE_ID = '5D7555DE-0C38-4FCC-BB54-C3C4A7E81201';

export const MAX_ATTACHMENT_BYTES = 30_000_000;
export const MAX_ATTACHMENT_LABEL = '30 MB';
